src/utils/module.c

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright (C) 2017  Red Hat, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Author: Vojtech Trefny <vtrefny@redhat.com>
+ */
+
+#include <glib.h>
+#include <libkmod.h>
+#include <string.h>
+#include <syslog.h>
+#include <locale.h>
+
+#include "module.h"
+
+
+/**
+ * bd_utils_module_error_quark: (skip)
+ */
+GQuark bd_utils_module_error_quark (void)
+{
+    return g_quark_from_static_string ("g-bd-utils-module-error-quark");
+}
+
+gboolean bd_utils_have_kernel_module (const gchar *module_name, GError **error) {
```

Comment:
Please add documentation for these new public functions and add them to the *docs/libblockdev-sections.txt* file.